datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -236,6 +236,10 @@ public class Coordinator implements ZkAdapter.ZkAdapterListener, MetricsAware {
   private final Lock _throughputViolatingTopicsMapWriteLock = _throughputViolatingTopicsMapReadWriteLock.writeLock();
   private final Lock _throughputViolatingTopicsMapReadLock = _throughputViolatingTopicsMapReadWriteLock.readLock();
 
+  // This CV helps to halt threads (zk callback threads, main server thread) before attempting to acquire the Coordinator
+  // object. We never halt the event thread (coordinator thread) explicitly via this CV.
+  private final Object _conditionalVariableForCoordinatorObjectSynchronization = new Object();
```

Comment:
I think another way to define condition variable in Java is to explicitly define a lock and create a CV out of it. Instead of using synchronized blocks on the object around the methods, we can use this lock explicitly. https://www.iitk.ac.in/esc101/05Aug/tutorial/essential/threads/explicitlocks.html#:~:text=To%20wait%20on%20an%20explicit,that%20the%20condition%20has%20occurred.